lib/ecto.ex

```diff
@@ -595,4 +595,21 @@ defmodule Ecto do
       true
     end
   end
+
+  @spec load(Ecto.Schema.t | map(), map() | Keyword.t | {list, list}) :: Ecto.Schema.t | map()
+  def load(schema_or_types, data) when is_map(data) do
+    load(schema_or_types, {Map.keys(data), Map.values(data)})
```

Comment:
The string key support in load is mainly for embeds that we get from database as a map with string keys (mongo also uses it as the basic loading mechanism, since we get maps back).